src/completion/widget.ts

```diff
@@ -149,20 +149,24 @@ class CompletionWidget extends Widget {
    * Handle `after_attach` messages for the widget.
    *
    * #### Notes
-   * Captures document events in capture phase to dismiss or navigate the
+   * Captures window events in capture phase to dismiss or navigate the
    * completion widget.
+   *
+   * Because its parent (reference) widgets use window listeners instead of
+   * document listeners, the completion widget must also use window listeners
+   * in the capture phase.
```

Comment:
Right, but if a child cell adds `document` listeners and then the completion widget *also* adds a `document` listener, they are in contention. The only way to guarantee that the completion widget has first dibs is to make it add a `window` listener and use the capture phase. I'm not satisfied completely with this solution because that allows for only two levels of priority: one for `document` and one for `window`. What if we need more? But for the time being, it achieves the right behavior.